### DISCUSSION TITLE
Events Propagating From Html Overlays ( modal etc ) to pixi viewport

### DISCUSSION THREAD

**Discussion by Amansaxena001** [pixijs/pixijs#10911] [Q&A]
There is a canvas on the top of that we have a html modal where in if we hover or perform and any actions like click those are propagating to pixi. Is there a workaround to stop this behaviour. I know that pixi registers the events on window due to which this is happening. 

Version - v7

**Comment by Amansaxena001** [comment]
Went through this [thread](https://github.com/pixijs/pixijs/issues/1725) too but no tangible solutions there

**Comment by Amansaxena001** [comment]
@ivanpopelyshev @themoonrat  ?  any workarounds to solve this? The issue is only with pointerMove that is getting triggered every time even when user is not hovering over viewport

**Reply by Amansaxena001** [reply]
@bigtimebuddy if you have some time

**Reply by bigtimebuddy** [reply]
When you open your modal, can you set `pointer-events: none` on the canvas? Does that do anything?

**Reply by Amansaxena001** [reply]
I did that still the `pointermove` was getting triggered but all the other events like click, pointerdown stopped working that was expected @bigtimebuddy 

Also the events are attached to `Viewport` from `pixi-viewport` 

"pixi-viewport": "5.0.2",

**Reply by bigtimebuddy** [reply]
Why don't you ask on the pixi-viewport project? Might be setting window events.

**Reply by Amansaxena001** [reply]
I ve asked them too but still wanted some opinions from pixi side if something can be done

**Reply by joakimlindblad** [reply]
Afaics, pointermove are nowadays (v8) captured in the Capture phase of event processing, already at the top (document) element.
`./events/EventSystem.ts:515:     globalThis.document.addEventListener('pointermove', this._onPointerMove, true);`

Thus, ignoring anything of rendering, z-index, or canvases (modals) on top. 

The only way I have found to externally prevent Pixi from grabbing the pointermove events is to stop them before.
This can be done with an event listener added **before** pixi is scripted.
```
function blockPointerMove(event) {
  event.stopImmediatePropagation();
}
document.addEventListener("pointermove", blockPointerMove, true); // capture phase
```

However, this blocks **all** pointermove events to the whole application. And, there seems to be no way to re-route events to lower elements in the DOM tree. Or as Copilot states it: 'Unfortunately, there's no native way in the DOM to dispatch an event that "skips" higher-level listeners like those on document. All dispatched events go through the full propagation path: capture → target → bubble.'

Thus, my best found way around it is to explicitly tell Pixi to ignore events, e.g., by explicating `container.eventMode = 'none';` when modal is active. This works OK for global covering objects, but is complicated if e.g., half the pixi-canvas is covered.

IMHO: The capturing of pointermove in the Capture phase at the topmost level is rather irritatingly forceful. Other events seem to be caught further down the tree, e.g. 
`./events/EventSystem.ts:516:     this.domElement.addEventListener('pointerdown', this._onPointerDown, true);`. 
Is the topmost capture really needed?

Has anyone found a better solution?
